jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py

```diff
@@ -64,10 +76,6 @@ def _on_viewer_data_changed(self, msg=None):
         self.dc_items = [lyr.label for lyr in self.app.data_collection
                          if layer_is_image_data(lyr)]
 
-        self.subset_items = [lyr.label for lyr in self.app.data_collection.subset_groups
-                             if lyr.label.startswith('Subset')]

```

Comment:
Ok, very good to know, I can move that logic into the component (at least until we support renaming subsets in which case we're going to have to rethink this...).

Reply:
Yeah, we need to rethink a lot of things, including this...

* #1141

Reply:
@pllim just to revisit this - are the arbitrary masks you mentioned loaded as subsets?  Can you give me an example so I can try to write other logic to filter them out from photometry?

Reply:
Yes, a bunch of those are loaded in the [Imviz example notebook](https://github.com/spacetelescope/jdaviz/blob/main/notebooks/ImvizExample.ipynb); look for the cell that says `imviz.load_static_regions(my_regions)`. In the example, all the names start with `my_` for simplicity but in reality, you can name it whatever you want.

Reply:
The unfortunate thing is in order to support arbitrary shapes, everything (even something that can be cleanly converted to something that is compatible with photometry aperture) is converted to a masked state that is incompatible. That is why I said we need to rethink life.

https://github.com/spacetelescope/jdaviz/blob/a9b151d1ca91af798fcfb83bfcf34d5a5d0cf1dd/jdaviz/configs/imviz/helper.py#L236

Reply:
Thanks, that's very helpful.  In exploring this scenario in #1291, it doesn't seem that the situation is quite as bad as we feared as other logic in the component will see these as non-spatial subsets (unfortunately that means it assumes they're spectral, but since this only happens in imviz, that distinction isn't so important).  It will definitely require more thought and testing though.